rs/tests/driver/src/k8s/tnet.rs

```diff
@@ -391,24 +396,51 @@ impl TNet {
                     format!("{}-{}", self.nodes.len(), vm_req.name),
             }
         );
-        let pvc_name = format!("{}-guestos", vm_name.clone());
-        let data_source = Some(TypedLocalObjectReference {
-            api_group: None,
-            kind: "PersistentVolumeClaim".to_string(),
-            name: match vm_req.primary_image {
-                ImageLocation::PersistentVolumeClaim { name } => name,
-                _ => unimplemented!(),
-            },
-        });
 
-        create_pvc(
-            &k8s_client.api_pvc,
-            &pvc_name,
-            "100Gi",
-            None,
-            None,
-            data_source,
+        let node = create_reservation(
+            vm_name.clone(),
+            vm_name.clone(),
+            self.unique_name.clone().expect("missing unique name"),
+            "1h".to_string().into(),
+            Some((
+                vm_req.vcpus.to_string(),
+                vm_req.memory_kibibytes.to_string() + "Ki",
+            )),
+        )
+        .await?;
+
+        // create a job to download the image and extract it
+        let image_url = format!(
+            "http://server.bazel-remote.svc.cluster.local:8080/cas/{}",
+            match vm_req.primary_image {
+                ImageLocation::IcOsImageViaUrl { url: _, sha256 } => sha256,
+                _ => self.image_sha.clone(),
+            }
+        );
+        // TODO: only download it once and copy it if it's already downloaded
+        let args = format!(
+            "set -e; apk add zstd tar; \
+            mkdir -p /tnet/{vm_name}; \
+            wget -O /tnet/{vm_name}/img.tar.zst {image_url}; \
+            unzstd -d /tnet/{vm_name}/img.tar.zst; \
+            tar -xvf /tnet/{vm_name}/img.tar -C /tnet/{vm_name}; \
```

Comment:
You could also let tar do the zstd uncompression like we [do in Farm](https://github.com/dfinity-lab/farm/blob/master/backend/Farm/VMManager.hs#L1188-L1201):
```suggestion
            tar -x --zstd -vf /tnet/{vm_name}/img.tar.zst -C /tnet/{vm_name}; \
```